Validation/HGCalValidation/src/HGVHistoProducerAlgo.cc

```diff
@@ -0,0 +1,1000 @@
+    unsigned int numberOfHaloHitsInLC = 0;
+
+    for (unsigned int hitId = 0; hitId < numberOfHitsInLC; hitId++)
+    {
+      DetId rh_detid = hits_and_fractions[hitId].first;
+      auto rhFraction = hits_and_fractions[hitId].second;
+
+      std::map<DetId,const HGCRecHit *>::const_iterator itcheck= hitMap.find(rh_detid);
+      const HGCRecHit *hit = itcheck->second;
+
+
+      auto hit_find_in_LC = detIdToLayerClusterId_Map.find(rh_detid);
+      if (hit_find_in_LC == detIdToLayerClusterId_Map.end())
+      {
+        detIdToLayerClusterId_Map[rh_detid] = std::vector<HGVHistoProducerAlgo::detIdInfoInCluster> ();
+      }
+      detIdToLayerClusterId_Map[rh_detid].emplace_back(HGVHistoProducerAlgo::detIdInfoInCluster{lcId,rhFraction});
+
+      auto hit_find_in_CP = detIdToCaloParticleId_Map.find(rh_detid);
+
+      // if the fraction is zero or the hit does not belong to any calo
+      // particle, set the caloparticleId for the hit to -1 this will
+      // contribute to the number of noise hits
+
+      // MR Remove the case in which the fraction is 0, since this could be a
+      // real hit that has been marked as halo.
+      if (rhFraction == 0.) {
+        hitsToCaloParticleId[hitId] = -2;
+        numberOfHaloHitsInLC++;
+      }
+      if (hit_find_in_CP == detIdToCaloParticleId_Map.end())
+      {
+        hitsToCaloParticleId[hitId] -= 1;
+      }
+      else
+      {
+        auto maxCPEnergyInLC = 0.f;
+        auto maxCPId = -1;
+        for(auto& h: hit_find_in_CP->second)
+        {
+          CPEnergyInLC[h.clusterId] += h.fraction*hit->energy();
+          cPOnLayer[h.clusterId][lcLayerId].layerClusterIdToEnergyAndScore[lcId].first += h.fraction*hit->energy();
+          cpsInLayerCluster[lcId].emplace_back(std::make_pair<int, float>(h.clusterId, 0.f));
+          if(h.fraction >maxCPEnergyInLC)
+          {
+            maxCPEnergyInLC = CPEnergyInLC[h.clusterId];
+            maxCPId = h.clusterId;
+          }
+        }
+        hitsToCaloParticleId[hitId] = maxCPId;
+      }
+      histograms.h_cellAssociation_perlayer.at(lcLayerId%52+1).fill(hitsToCaloParticleId[hitId] > 0. ? 0. : hitsToCaloParticleId[hitId]);
+    }
+
+    for(auto& c: hitsToCaloParticleId)
+    {
+      if(c < 0)
+      {
+        numberOfNoiseHitsInLC++;
+      }
+      else
+      {
+        occurrencesCPinLC[c]++;
+      }
+    }
+
+    for(auto&c : occurrencesCPinLC)
+    {
+      if(c.second > maxCPNumberOfHitsInLC)
+      {
+        maxCPId_byNumberOfHits = c.first;
+        maxCPNumberOfHitsInLC = c.second;
+      }
+    }
+
+    for(auto&c : CPEnergyInLC)
+    {
+      if(c.second > maxEnergySharedLCandCP)
+      {
+        maxCPId_byEnergy = c.first;
+        maxEnergySharedLCandCP = c.second;
+      }
+    }
+    float totalCPEnergyOnLayer = 0.f;
+    if(maxCPId_byEnergy >=0) {
+      totalCPEnergyOnLayer = cPOnLayer[maxCPId_byEnergy][lcLayerId].energy;
+      energyFractionOfCPinLC = maxEnergySharedLCandCP/totalCPEnergyOnLayer;
```

Comment:
this line generates a warning in the static analyzer
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26099/33460/llvm-analysis/report-b756de.html#EndPath

There are 4 more https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26099/33460/llvm-analysis/

Reply:
It seems that setting and using these variables only for couts produces these warnings. We would like to leave the information in the code in order to use it when we will need it for debugging purposes, if this is not a big problem.